src/evict/evict_page.c

```diff
@@ -822,20 +822,20 @@ __evict_reconcile(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags)
 
     /*
      * Acquire a snapshot if coming through the eviction thread route. Also, if we have entered
-     * eviction through application threads and we have a transaction snapshot, we will use our
-     * existing snapshot to evict pages that are not globally visible based on the last_running
-     * transaction. Avoid using snapshots when application transactions are in the final stages of
-     * commit or rollback as they have already released the snapshot. Otherwise, it becomes harder
-     * in the later part of the code to detect updates that belonged to the last running application
-     * transaction.
+     * eviction through application threads then we take a backup of the existing snapshot and
+     * acquire a new snapshot, once the application threads are done with eviction then we switch
+     * back the snapshot to its original. Avoid using snapshots when application transactions are in
+     * the final stages of commit or rollback as they have already released the snapshot. Otherwise,
+     * it becomes harder in the later part of the code to detect updates that belonged to the last
+     * running application transaction.
      */
-    use_snapshot_for_app_thread = !F_ISSET(session, WT_SESSION_INTERNAL) &&
+    is_application_thread = !F_ISSET(session, WT_SESSION_INTERNAL) &&
```

Comment:
May be rename it to bump_app_snapshot? We have the check for F_ISSET(session->txn, WT_TXN_HAS_SNAPSHOT) so this only applies to app thread with a snapshot.

Reply:
I named the variable specifically `is_application_thread ` so that it is clear that the application thread is doing the eviction and there also exists a function `__wt_txn_bump_snapshot` that is called in eviction flow so I think `bump_app_snapshot` would create confusion.

Reply:
Then may be is_application_thread_with_snapshot?

Reply:
Fixed in ecda3b0507dd490543c80a6018c3289d814672cf